sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeExtractors.scala

```diff
@@ -343,6 +344,58 @@ case class GpuArrayContains(left: Expression, right: Expression)
   override def prettyName: String = "array_contains"
 }
 
+case class GpuArrayPosition(left: Expression, right: Expression)
+  extends GpuBinaryExpression with NullIntolerantShim {
+
+  override def dataType: DataType = LongType
+
+  override def prettyName: String = "array_position"
+
+  override def doColumnar(numRows: Int, lhs: GpuScalar, rhs: GpuScalar): ColumnVector = {
+    withResource(GpuColumnVector.from(lhs, numRows, lhs.dataType)) { left =>
+      withResource(GpuColumnVector.from(rhs, numRows, rhs.dataType)) { right =>
```

Comment:
Here can expand only the left scalar to a column and leverage the version of (column, scalar) to reduce memory pressure.